src/core/mesh/qgsmeshadvancedediting.h

```diff
@@ -124,4 +125,67 @@ class CORE_EXPORT QgsMeshEditRefineFaces : public QgsMeshAdvancedEditing
 };
 
 
+/**
+ * \ingroup core
+ *
+ * \brief Class that can transform vertices of a mesh by expression
+ *
+ * Each coordinates are associated with an expression that can be defined with function
+ * returning the current coordinates (see setExpressions()):
+ *
+ * - $vertex_x
+ * - $vertex_y
+ * - $vertex_z
+ *
+ * Example:
+ * Transposing a mesh and translate following axe X with a distance of 50 and increase the level of the mesh
+ * with an height of 80 when previous X coordinate is under 100 and de crease the level of 150 when X is under 100:
+ *
+ * expressionX: "$vertex_y + 50"
+ * expressionY: "$vertex_x"
+ * expressionZ: "if( $vertex_x <= 100 , $vertex_z + 80 , $vertex_z - 150)"
+ *
+ * \since QGIS 3.22
+ */
+class CORE_EXPORT QgsMeshTransformVerticesByExpression : public QgsMeshAdvancedEditing
+{
+  public:
+
+    //! Constructor
+    QgsMeshTransformVerticesByExpression() = default;
+
+    /**
+     * Sets the expressions for the coordinates transformation.
+     *
+     * \note Expressions are optional for each coordinate, the coordinte will ot be transformed if the string is void.
+     */
+    void setExpressions( const QString &expressionX, const QString &expressionY, const QString &expressionZ );
+
+    /**
+     * Calculates the transformed vertices of the mesh \a layer, returns FALSE if this leads to topological or geometrical errors.
+     * The mesh layer must be in edit mode.
+     *
+     * \note this method not apply new vertices to the mesh layer but only store the calculated transformation
+     *       that can be apply later with apply()
```

Comment:
apply is private, so as a consumer of this class, I am not sure how should I use the class API?

Reply:
true, the user have to use `QgsMesEditor::advanceEdit(QgsMeshAdvancedEdit *edit)`, doc updated